klaviyo/api_helper.py

```diff
@@ -200,6 +204,20 @@ def _public_request(self, path, querystring):
         url = '{}/{}?{}'.format(self.api_server, path, querystring)
         return self._request(self.HTTP_GET, url, request_type=self.PUBLIC)
 
+    def _public_post_request(self, url, datastring, headers):
+        """Track and identify calls, always a get request.
+
+        Args:
+            path (str): track or identify.
+            querystring (str): urlencoded & b64 encoded string.
+        Returns:
+            (str): 1 or 0 (pass/fail).
+        """
+
+        response = requests.request("POST", url, data=datastring, headers=headers)
```

Comment:
why can't you just use the `_request` method? This does this for you, maybe can just pass the headers you want which look like it's the same for post requests here rather than typing them out multiple times

Reply:
i can't use _request, because that method hardcodes the headers to be application/json , which is not what we want

Reply:
an opportunity for a refactor - that should be added as a kwarg and you can dynamically pass that in!

Reply:
this should just be `return self._request(.....)`

Reply:
refactored _request